added profile component files

JyothiNandyala opened the pull request:
# Description

Storybook Profile Form Component

Fixes # (issue)
#43 
# How to test?

Please provide a short summary how your changes can be tested?

# Checklist

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] This PR is ready to be merged and not breaking any other functionality

TITHI1244 commented on src/client/components/ProfileFormComponent/ProfileFormComponent.css:
```diff
@@ -0,0 +1,73 @@
+
+.form-With-Avatar {
+  height: 100%;
+  width: 200px;
```
I think, we do not need this width as we are already mentioning one default width of 400px.

JyothiNandyala replied:
yes tithi here width is not required i removed it.

TITHI1244 commented on src/client/components/ProfileFormComponent/ProfileFormComponent.css:
```diff
@@ -0,0 +1,73 @@
+.container .edit-Link a {
+  text-decoration: none;
+  font-size: 20px;
+}
```
We can add some width to the .container, otherwise it's inheriting width from its parent and so breaking. Tried with this one, worked fine I think :)
`.container {
  width: 85%;
}`

JyothiNandyala replied:
yes tithi i gave the width for entite component so now its working fine.

TITHI1244 commented on src/client/components/ProfileFormComponent/ProfileFormComponent.css:
```diff
@@ -0,0 +1,73 @@
+  display: flex;
+  margin: 40px auto;
+  justify-content: center;
+  width: 280px;
```
I guess, we can use some % instead of px value to width so that it doesn't break :) Tried the following, please check.
`width: 60%;
  max-width: 250px;`

TITHI1244 commented on src/client/components/ProfileFormComponent/ProfileFormComponent.stories.js:
```diff
@@ -0,0 +1,29 @@
+  component: Avatar,
+  argTypes: {
+    color: {
+      control: { type: 'color' },
```
I guess, this control is supposed to be select rather than type:color. As the docs says, there would be 3 options of colors only not the whole spectrum :) 

> COLORS #DB643D, #535C75 and # 000000

JyothiNandyala replied:
yes i changed the color type to select.

TITHI1244 commented on src/client/components/ProfileFormComponent/ProfileFormComponent.stories.js:
```diff
@@ -0,0 +1,29 @@
+    },
+    width: {
+      control: { type: 'range', min: 400, max: 1200, step: 50 },
+    },
```
Don't we need one more control for fonts?

JyothiNandyala replied:
yes tithi in the docs we have fonts also but when iam using in storybook i can't see any difference.maybe i will check that

JyothiNandyala replied:
yes tithi we need fonts control also i did that to.

TITHI1244 reviewed (commented):
Nice work @JyothiNandyala . Suggested a few changes, please check if you think same as me :) Thanks.

valmdz reviewed (changes requested):
good job Jyothi, I added some comments. I believe you do not need to push the changes made on _package-lock.json_ as they are more to be kept in your local, unless you have installed something for the project that everyone else should have, but in that case it is better that you make a separate PR for that :)

valmdz commented on src/client/index.css:
```diff
@@ -2,7 +2,7 @@
 
 * {
-  font-family: 'Montserrat', sans-serif;
+  /* font-family: 'Montserrat', sans-serif; */
```
Also remember to remove all your unused code

JyothiNandyala replied:
ya i removed it.

JyothiNandyala commented on src/client/components/ProfileFormComponent/ProfileFormComponent.css:
```diff
@@ -0,0 +1,72 @@
+  margin: 40px auto;
+  justify-content: center;
+  height: 198px;
+  width: 290px;
```
ya ok ,now iam using only width for the image.

JyothiNandyala commented on src/client/components/ProfileFormComponent/ProfileFormComponent.js:
```diff
@@ -0,0 +1,55 @@
+      <div className="avatar-Image">
+        <figure>
+          <img src={avatarimage} alt="avatarimage" />
+          <figcaption>Anne</figcaption>
```
Hi valmdz i changed those name,image,email and password to props and now user can able to change the users information,could you please let me know is am doing wrong/right.

soujanyapolavarapu commented on src/client/components/ProfileFormComponent/ProfileFormComponent.stories.js:
```diff
@@ -0,0 +1,64 @@
+import React from 'react';
+import Avatar from './ProfileFormComponent';
+import avatarImage from '../../assets/images/avatar.png';
```
@JyothiNandyala , As i can see Cannot find module '../../assets/images/avatar.png' or its corresponding type declarations. (may be it is a warning)

JyothiNandyala replied:
Hi soujanya ,i checked this in other person system's also they are not getting any error like '../../assets/images/avatar.png', may be could you please recheck and let me know.

soujanyapolavarapu commented on src/client/components/ProfileFormComponent/ProfileFormComponent.stories.js:
```diff
@@ -0,0 +1,64 @@
+  userPassword,
+  userImage,
+}) => (
+  <Avatar
```
here i can see this :   Property 'onChange' is missing in type '{ color: any; width: any; fontFamily: any; userName: any; userEmail: any; userPassword: any; userImage: any; }' but required in type 'Pick<Pick<{ color: any; width: any; fontFamily: any; userName: any; userEmail: any; userPassword: any; onChange: any; userImage: any; }, never> & Pick<InferProps<{ color: Requireable<string>; width: Validator<...>; fontFamily: Requireable<...>; }>, "color" | ... 1 more ... | "fontFamily"> & Pick<...>, "width" | ... ...'.
(If it is warning you can ignore)

JyothiNandyala replied:
because i was not added those variables in proptypes , i will do that thing, thank you.

soujanyapolavarapu reviewed (approved):
Good Job Jyothi :) I just pulled you branch and it showed me that comments which i have commented.  :)

JyothiNandyala commented:
> good job Jyothi, I added some comments. I believe you do not need to push the changes made on _package-lock.json_ as they are more to be kept in your local, unless you have installed something for the project that everyone else should have, but in that case it is better that you make a separate PR for that :)



> good job Jyothi, I added some comments. I believe you do not need to push the changes made on _package-lock.json_ as they are more to be kept in your local, unless you have installed something for the project that everyone else should have, but in that case it is better that you make a separate PR for that :)

> good job Jyothi, I added some comments. I believe you do not need to push the changes made on _package-lock.json_ as they are more to be kept in your local, unless you have installed something for the project that everyone else should have, but in that case it is better that you make a separate PR for that :)

ya ok thank you.

dpfernandes reviewed (approved):
👍
